src/queries/Vesla/getObservationsQuery.ts

```diff
@@ -60,8 +65,11 @@ export async function getObservationSiteIds(
   params: CommonParameters,
   obsCode: string
 ) {
-  const filter = await getFilter(params, obsCode)
-  let data = await getVeslaData(resource, '$select=siteId' + filter)
+  const filter = getFilter(params, obsCode)
+  let data = await getVeslaData(
+    resource,
+    '$apply=filter(' + filter + ')/groupby((siteId))&$orderby=siteId'
```

Comment:
So the actual functional change here is to use `$apply` with `groupby` which ensures we only get unique site ids